pyproject.toml

```diff
@@ -88,6 +88,57 @@ testpaths = [
     "tests",
 ]
 
+[tool.ruff]
+fix = true
+force-exclude = true
+ignore = [
+    "B905", # python>=3.10
+    "EM101",
+    "EM102",
+    "N802",
+    "N803",
+    "N806",
+    "S101",
+    "UP006",
+    "UP007",
+    "UP015",
```

Comment:
Which ones have you ignored here?

Reply:
I'm not sure what you mean? Most of these ignores were also in the previous linting pr. I've added `UP006`, `UP007`, and `UP015` - the first two because these are to do with type hints - using e.g. `Union` instead of `|` and `List` instead of `list` - and there are lots of places where we have to use Union and List anyway. And `UP015` is about using not needing to use `mode='r'` when opening a file as it's the default, but there's no harm in being explicit

Reply:
Oh, I'm just genuinely curious. Hard to tell when looking at something "done" by someone else.